src/Interpreters/ClusterProxy/SelectStreamFactory.cpp

```diff
@@ -120,17 +122,41 @@ void SelectStreamFactory::createForShard(
     auto it = objects_by_shard.find(shard_info.shard_num);
     if (it != objects_by_shard.end())
         replaceMissedSubcolumnsByConstants(storage_snapshot->object_columns, it->second, query_ast);
+    
+    const auto & settings = context->getSettingsRef();    
+    ASTPtr query;
+    
+    if (settings.max_execution_time_leaf.value != 0)
+    {
+        query = query_ast->clone();
+        /// Replace 'max_execution_time' of this sub-query with 'max_execution_time_leaf' and 'timeout_overflow_mode'
+        /// with 'timeout_overflow_mode_leaf'
+        if (auto * select = query->as<ASTSelectQuery>())
+        {
+            auto query_settings = select->settings();
+            if (!query_settings)
+                query_settings = std::make_shared<ASTSetQuery>();
+            
+            query_settings->as<ASTSetQuery &>().is_standalone = false;
```

Comment:
Interesting... Usually, we send the modified settings in the server protocol, not inline in the query. I will check if this method of sending settings is viable.

Reply:
Why not to change these settings in [updateSettingsForCluster](https://github.com/ClickHouse/ClickHouse/blob/aa7c1f63abda4cdb8a592c9d90dc3a7d2ad3a935/src/Interpreters/ClusterProxy/executeQuery.cpp#L38)? Is there a reason why we cannot do it there?

Reply:
@Avogar I changed the implementation as your suggestion.